cardano-crypto-class/src/Cardano/Crypto/DSIGN/Ed25519.hs

```diff
@@ -0,0 +1,105 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- | Ed25519 digital signatures.
+module Cardano.Crypto.DSIGN.Ed25519
+  ( Ed25519DSIGN
+  , SigDSIGN (..)
+  , SignKeyDSIGN (..)
+  , VerKeyDSIGN (..)
+  )
+where
+
+import Cardano.Binary
+  ( Decoder
+  , Encoding
+  , FromCBOR (..)
+  , ToCBOR (..)
+  , serialize
+  )
+import Cardano.Crypto.DSIGN.Class
+import Cardano.Prelude (NoUnexpectedThunks, UseIsNormalForm(..))
+import Crypto.Error (CryptoFailable (..))
+import Crypto.PubKey.Ed25519
+import Data.ByteArray (ByteArrayAccess, convert)
+import Data.ByteString (ByteString)
+import Data.ByteString.Lazy (toStrict)
+import GHC.Generics (Generic)
+
+data Ed25519DSIGN
+
+instance DSIGNAlgorithm Ed25519DSIGN where
+
+    type Signable Ed25519DSIGN = ToCBOR
```

Comment:
This is not right for the real thing. We must never re-serialise for verification. Instead this should be the class to recover the annotated bytes that we already have.

Reply:
The new Shelley annotations work in this way. The `ToCBOR` instance does _not_ serialise but instead uses the annotation